### PR TITLE
Added Token Type Filter in search_assets fuction

### PR DESCRIPTION
**Pull request by AhzamAkhtar**
Types of filters that are added in search_assets function are
- [x] Fungible
- [x] NonFungible
- [x] Compressed
- [x] Nft
- [x] All
- [x] Tests

### CODE REVIEW

File: digital_asset_types/src/dao/mod.rs [hunk 1]
```diff
@@ -93,6 +96,23 @@ impl SearchAssetsQuery {
                     .clone()
                     .map(|x| asset::Column::SpecificationAssetClass.eq(x)),
             )
+            .add_option(
```
Comment (Nagaprasadvr) [inline digital_asset_types/src/dao/mod.rs:99]: i guess we can remove specification_asset_class option now as we have token_type

File: digital_asset_types/src/dao/mod.rs [hunk 2]
```diff
@@ -54,6 +56,7 @@ pub struct SearchAssetsQuery {
     pub condition_type: Option<ConditionType>,
     pub specification_version: Option<SpecificationVersions>,
     pub specification_asset_class: Option<SpecificationAssetClass>,
+    pub token_type: Option<TokenTypeClass>,
```
Comment (Nagaprasadvr) [inline digital_asset_types/src/dao/mod.rs:59]: remove SpecificationAssetClass option

**Review by Nagaprasadvr — COMMENTED** [review]
looks good , we can remove specification_asset_class in the query params

File: digital_asset_types/src/dao/mod.rs (hunk 1 above)
Comment (AhzamAkhtar) [reply digital_asset_types/src/dao/mod.rs:99]: yeah think so.

**Review by kespinola — CHANGES_REQUESTED** [review]
Looks good but needs integration tests @AhzamAkhtar

**Comment by AhzamAkhtar** [comment]
> Looks good but needs integration tests @AhzamAkhtar

Integration tests added @kespinola

**Review by kespinola — APPROVED** [review]

**Pull request closed without merging**